hope_horizon/backend/views/group_view.py

```diff
@@ -0,0 +1,94 @@
+from rest_framework import viewsets, status
+from rest_framework.response import Response
+from backend.serializers import GroupSerializer
+from backend.models import CustomGroup, GroupUser
+from rest_framework.permissions import IsAuthenticated
+from django.db.models import Q
+
+class GroupViewSet(viewsets.ModelViewSet):
+    permission_classes = [IsAuthenticated]
+    serializer_class = GroupSerializer
+
+    def list(self, request):
+        user = request.user
+        owned = request.query_params.get("owned", None)
+        if owned is None:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+        if owned.lower() == "true":
+            group_user_owned = GroupUser.objects.filter(user=user).filter(is_owner=True).filter(is_active=True)
+            group = CustomGroup.objects.filter(id__in=group_user_owned.values_list("group", flat=True))
+        else:
+            group_user_member = GroupUser.objects.filter(user=user).filter(is_owner=False).filter(is_active=True)
+            group = CustomGroup.objects.filter(id__in=group_user_member.values_list("group", flat=True))
+        serializer = self.serializer_class(group, many=True)
+        return Response({"groups": serializer.data}, status=status.HTTP_200_OK)
+    
+    def retrieve(self, request, pk=None):
+        user = request.user
+        try:
+            group = CustomGroup.objects.get(pk=pk)
+        except CustomGroup.DoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+        if group.is_active == False:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+        #check if the user is a member of the group and is active
+        try:
+            GroupUser.objects.filter(user=user).filter(group=group).filter(is_active=True).get()
+        except GroupUser.DoesNotExist:
+            return Response(status=status.HTTP_403_FORBIDDEN) 
+        serializer = self.serializer_class(group)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+    
+
+    def create(self, request):
+        serializer = self.get_serializer(data=request.data)
+        if request.data.get("name") is None or request.data.get("description") is None:
+            return Response("Invalid request data", status=status.HTTP_400_BAD_REQUEST)
+        serializer.is_valid()
```

Comment:
you should also handle when the serializer is not valid.
example:
if not serializer.is_valid():
            return Response({"detail": serializer.errors}, status=status.HTTP_400_BAD_REQUEST)